### PR TITLE
CATROID-115 Import projects

### PR DESCRIPTION
**Pull request by thmq**
- fixed issues with project migration from external storage.

NOTE: I also changed the method `setProject(..)` to `setCurrentProject(..)` in `Projectmanager.java`, because the corresponding methods for current scene, sprite, etc. are also called `setCurrentWHATEVERITISYOUWANTTOSET(..)`. this method hat a lot of usages (i.e. 169) that's why so many files have changed.

### CODE REVIEW

**Comment by LPeteR90** [comment]
Currently there is a checkstyle failure in ProjectListFragment.java:39 (Import order)

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectImportTask.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		for (File projectDir : files) {
+			success = success && importProject(projectDir);
+		}
+		return success;
+	}
+
+	@Override
+	protected void onPostExecute(Boolean success) {
+		if (weakListenerReference.get() != null) {
+			weakListenerReference.get().onImportFinished(success);
```
Comment (LPeteR90) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectImportTask.java]: `weakReferenceListener.get()` potentially null

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectImportTask.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+				fileNames.add(dir.getName());
+			}
+
+			if (dstDir.exists()) {
+				File newDstDir = new File(DEFAULT_ROOT_DIRECTORY,
+						new UniqueNameProvider().getUniqueName(projectDir.getName(), fileNames));
+				try {
+					StorageOperations.copyDir(projectDir, newDstDir);
+					Project copiedProject = XstreamSerializer.getInstance()
+							.loadProject(newDstDir.getName(), weakContextReference.get());
```
Comment (LPeteR90) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectImportTask.java]: `weakContextReference.get()` potentially null

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectLoadTask.java [hunk 3]
```diff
@@ -25,39 +25,45 @@
...
+	protected Boolean doInBackground(String... strings) {
+		if (weakContextReference.get() == null) {
+			return false;
+		}
 		try {
-			ProjectManager.getInstance().loadProject(strings[0], context);
-			return "";
-		} catch (ProjectException e) {
-			return e.getUiErrorMessage();
+			ProjectManager.getInstance().loadProject(strings[0], weakContextReference.get());
```
Comment (LPeteR90) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectLoadTask.java]: `weakContextReference.get()` potentially null

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectLoadTask.java [hunk 4]
```diff
@@ -25,39 +25,45 @@
...
+			return false;
 		}
 	}
 
 	@Override
-	protected void onPostExecute(String s) {
-		listener.onLoadFinished(s.equals(""), s);
+	protected void onPostExecute(Boolean success) {
+		if (weakListenerReference.get() != null) {
+			weakListenerReference.get().onLoadFinished(success);
```
Comment (LPeteR90) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectLoadTask.java]: `weakListenerReference.get()` potentially null

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectRenameTask.java [hunk 5]
```diff
@@ -25,31 +25,45 @@
...
 
 	@Override
-	protected Boolean doInBackground(Void... voids) {
-		return ProjectManager.getInstance().initializeDefaultProject(context);
+	protected Boolean doInBackground(String... strings) {
+		if (weakContextReference.get() == null) {
+			return false;
+		}
+		try {
+			ProjectManager.getInstance().renameProject(strings[0], strings[1], weakContextReference.get());
```
Comment (LPeteR90) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectRenameTask.java]: `weakContextReference.get()` potentially null

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectRenameTask.java [hunk 6]
```diff
@@ -25,31 +25,45 @@
...
+			return false;
+		}
 	}
 
 	@Override
-	protected void onPostExecute(Boolean aBoolean) {
-		listener.onCreateFinished(aBoolean);
+	protected void onPostExecute(Boolean success) {
+		if (weakListenerReference.get() != null) {
+			weakListenerReference.get().onRenameFinished(success);
```
Comment (LPeteR90) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectRenameTask.java]: `weakContextReference.get()` potentially null

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectCopyTask.java [hunk 7]
```diff
@@ -31,42 +31,48 @@
...
+
+		File projectToCopyDir = new File(DEFAULT_ROOT_DIRECTORY, strings[0]);
+		File projectDir = new File(DEFAULT_ROOT_DIRECTORY, strings[1]);
 
 		try {
-			StorageOperations.copyDir(projectToCopyDirectory, projectDirectory);
-			Project project = XstreamSerializer.getInstance().loadProject(params[1], context);
-			project.setName(params[1]);
+			StorageOperations.copyDir(projectToCopyDir, projectDir);
+			Project project = XstreamSerializer.getInstance().loadProject(strings[1], weakContextReference.get());
```
[the remark this answers is not in the document]
Comment (thmq) [reply catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/asynctask/ProjectCopyTask.java]: yes! excellent catch! thank you :+1:

**Pull request merged**